plugins/searchonline.koplugin/main.lua

```diff
@@ -0,0 +1,313 @@
+
+                        if true then
+                            -- Check for duplicates
+                            for _, existing_url in ipairs(self.user_search_engines) do
+                                if existing_url == input_text then
+                                    UIManager:show(InfoMessage:new{
+                                        text = _("This URL already exists"),
+                                    })
+                                    return
+                                end
+                            end
+
+                            -- Add the new URL
+                            table.insert(self.user_search_engines, input_text)
+                            table.insert(self.search_engines, input_text)
+                            G_reader_settings:saveSetting("searchonline_user_search_engines", self.user_search_engines)
+
+                            UIManager:show(InfoMessage:new{
+                                text = _("Search engine added successfully: " .. input_text),
+                            })
+                        end
+
+                        self.user_input:onCloseKeyboard()
+                        UIManager:close(self.user_input)
+                    end,
+                },
+            }
+        },
+    }
+    UIManager:show(self.user_input)
+    self.user_input:onShowKeyboard()
+end
+
+-- Look up words from the menu instead
+function SearchOnline:lookupInput()
+    self.input_dialog = InputDialog:new{
+        title = _("Enter a word or phrase to search online"),
+        input = "",
+        input_type = "text",
+        buttons = {
+            {
+                {
+                    text = _("Cancel"),
+                    id = "close",
+                    callback = function()
+                        UIManager:close(self.input_dialog)
+                    end,
+                },
+                {
+                    text = _("Search Online"),
+                    is_enter_default = true,
+                    callback = function()
+                        if self.input_dialog:getInputText() == "" then return end
+                        NetworkMgr:runWhenOnline(function()
+                            local query = util.cleanupSelectedText(self.input_dialog:getInputText())
+                            UIManager:close(self.input_dialog)
+                            self:chooseSearch(query)
+                        end)
+                    end,
+                },
+            }
+        },
+    }
+    UIManager:show(self.input_dialog)
+    self.input_dialog:onShowKeyboard()
+end
+
+function SearchOnline:addToMainMenu(menu_items)
+    menu_items.searchonline_lookup = {
+        text = _("Search online"),
+        sorting_hint = "search",
+        callback = function() self:lookupInput() end,
+    }
+    menu_items.searchonline_restore_default_search_engines = {
+        text = _("Restore default search engines"),
+        sorting_hint = "search",

```

Comment:
This one and the next should probably not be in the first-level Search menu (to not make it more crowded than it already is with rarely used items), but in its Settings submenu.